flink-ml-benchmark/src/test/java/org/apache/flink/ml/benchmark/BenchmarkTest.java

```diff
@@ -0,0 +1,114 @@
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.Map;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/** Tests benchmarks. */
+@SuppressWarnings("unchecked")
+public class BenchmarkTest extends AbstractTestBase {
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    @Test
+    @SuppressWarnings({"unchecked"})
+    public void testParseJsonFile() throws Exception {
+        File configFile = new File(tempFolder.newFolder().getAbsolutePath() + "/test-conf.json");
+        InputStream inputStream =
+                this.getClass().getClassLoader().getResourceAsStream("benchmark-conf.json");
+        FileUtils.copyInputStreamToFile(inputStream, configFile);
+
+        Map<String, ?> benchmarks = BenchmarkUtils.parseJsonFile(configFile.getAbsolutePath());
+        assertEquals(benchmarks.size(), 1);
+        assertTrue(benchmarks.containsKey("KMeansModel-1"));
+
+        Map<String, ?> benchmark = (Map<String, ?>) benchmarks.get("KMeansModel-1");
+
+        KMeansModel stage =
+                ReadWriteUtils.instantiateWithParams((Map<String, ?>) benchmark.get("stage"));
+        assertEquals("prediction", stage.getPredictionCol());
+        assertEquals("features", stage.getFeaturesCol());
+        assertEquals(2, stage.getK());
+        assertEquals(EuclideanDistanceMeasure.NAME, stage.getDistanceMeasure());
+
+        DenseVectorGenerator inputDataGenerator =
+                ReadWriteUtils.instantiateWithParams((Map<String, ?>) benchmark.get("inputData"));
+        assertArrayEquals(new String[] {"features"}, inputDataGenerator.getColNames());
+        assertEquals(2, inputDataGenerator.getSeed());
+        assertEquals(10000, inputDataGenerator.getNumValues());
+        assertEquals(10, inputDataGenerator.getVectorDim());
+
+        KMeansModelDataGenerator modelDataGenerator =
+                ReadWriteUtils.instantiateWithParams((Map<String, ?>) benchmark.get("modelData"));
+        assertEquals(1, modelDataGenerator.getSeed());
+        assertEquals(2, modelDataGenerator.getArraySize());
+        assertEquals(10, modelDataGenerator.getVectorDim());
+    }
+
+    @Test
+    public void testRunBenchmark() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+        KMeans kMeans = new KMeans().setK(5).setFeaturesCol("test_feature");
+        InputDataGenerator<?> inputsGenerator =
+                new DenseVectorGenerator()
+                        .setColNames("test_feature")
+                        .setNumValues(1000L)
+                        .setVectorDim(10);
+
+        long estimatedTime = System.currentTimeMillis();
+        BenchmarkResult result =
+                BenchmarkUtils.runBenchmark(tEnv, "testBenchmarkName", kMeans, inputsGenerator);
```

Comment:
This method is only used in the test.

Could it be simpler to test the method that is actually used in the production code, e.g. `runBenchmark(StreamTableEnvironment tEnv, String name, Map<String, ?> params)`?